processing/src/main/java/org/apache/druid/segment/nested/NestedFieldColumnIndexSupplier.java

```diff
@@ -119,8 +121,8 @@ public NestedFieldColumnIndexSupplier(
     this.arrayElementBitmaps = arrayElementBitmaps;
     this.adjustLongId = globalStringDictionarySupplier.get().size();
     this.adjustDoubleId = adjustLongId + globalLongDictionarySupplier.get().size();
-    this.skipRangeIndexThreshold = (int) Math.ceil(skipValueRangeIndexScale * numRows);
-    this.skipPredicateIndex = localDictionarySupplier.get().size() > Math.ceil(skipValuePredicateIndexScale * numRows);
+    this.skipRangeIndexThreshold = (int) Math.ceil(columnConfig.skipValueRangeIndexScale() * numRows);
+    this.skipPredicateIndex = localDictionarySupplier.get().size() > Math.ceil(columnConfig.skipValuePredicateIndexScale() * numRows);
```

Comment:
I think it's a bit nicer to make these methods instead of fields that are computed on every constructor.  The computation that's being done is minimal, that's true, but it's better to only do it when needed.  I think that if you just store the reference to the `ColumnConfig` you can pretty easily move this check into private methods.  All of the places where you would check for skipping the predicate index, you would probably also be grabbing a `localDictionarySupplier` anyway, so you can reuse that singular reference too.